src/components/Borrow/BorrowInfo/BorrowInfo.tsx

```diff
@@ -0,0 +1,186 @@
+import { Grid, Popover, Typography } from '@material-ui/core'
+import React from 'react'
+import Copy from '@static/svg/copy.svg'
+import useStyles from './style'
+import { PublicKey } from '@solana/web3.js'
+interface IGeneralInfo {
+  collateralAmount: number
+  debtAmount: number
+  collateral: string
+  borrowed: string
+  limit: number
+  reserve: number
+  collateralAddress: PublicKey
+  borrowedAddress: PublicKey
+}
+
+export const BorrowInfo: React.FC<IGeneralInfo> = ({
+  collateral,
+  collateralAddress,
+  collateralAmount,
+  borrowed,
+  borrowedAddress,
+  reserve,
+  limit,
+  debtAmount
+}) => {
+  const classes = useStyles()
+  const [anchorEl, setAnchorEl] = React.useState<HTMLImageElement | null>(null)
+  const [open, setOpen] = React.useState(false)
+  const openCopiedText = (event: React.MouseEvent<HTMLImageElement, MouseEvent>) => {
+    if (event.currentTarget.id === 'collateral') {
```

Comment:
maybe instead of using id you can create higher-order function with text to copy as an argument